utils/mock-data-loader/tfm/api.js

```diff
@@ -63,7 +65,7 @@ const createTfmUser = async (user, token) => {
       'x-api-key': TFM_API_KEY,
     },
     url: `${TFM_API_URL}/v1/users`,
-    data: user,
+    data: { azureOid: user.username, ...user },
```

Comment:
Good spot, `azureOid` should be `oid` from entra user data, we have this in `mapEntraUserData` (outside of mock-data-loader)

Unless there's a reason for this @avaitonis ?

I would also change this so that the user destructuring comes first, otherwise the provided user object could be overriding`azureOid`. In fact, mabe we don't need `azureOid` here at all?

I think we should change to this:

```js
 data: { ...user, azureOid: user.oid },
```

But I would prefer this:

```js
 data: user,
```
Where the provided `user` would be a valid mock TFM user, with an `azureOid`. So the API call is minimal and doesn't require any mapping? Not sure how much of a change this would entail.

Thoughts?

Reply:
There's some thoughts around this where we want to do this `user` object creation -- if we want `tfm user` to always have these fields updaqting the mock does seem the best. if we dont, we could derive user tests cases from each other ala `portal-api\test-helpers\unit-test-mocks\mock-user.js` (note - this is my code and i dont particularly like it, but it does the job of being able to grab a user in different states thoughout the service for testing)

Reply:
I;m happy here -- if @ttbarnes is please feel free to resolve

Reply:
In TFM mock users we don't have `oid`, file utils\mock-data-loader\tfm\mocks\users.js .
In this PR we are introducing new required field `azureOid`, but we don't use it much, it just sounded as, maybe, good idea to save external SSO id in our DB. Maybe use it later for mapping users, support email change.

Because `azureOid` is new unique field and field `username` is unique, maybe it is fine to match them for test data?

@abhi-markan and everyone else, do we wan't to save azureOid in MongoDB at all?